### PR TITLE
it's `shasum -a 256` on darwin

### PR DESCRIPTION
**Pull request by auhlig**
@majewsky

### CODE REVIEW

File: golangvend [hunk 1]
```diff
@@ -38,7 +38,12 @@ CACHE_PATH="${PWD}/.golangvend-cache"
 
 # Given a string, print its SHA256 checksum.
 hash_string() {
-  echo "${1}" | sha256sum | cut -d' ' -f1
+  if [[ uname == "darwin"||"Darwin" ]]; then
```
Comment (majewsky) [inline golangvend:41]: Can you check whether this works on Linux? This doesn't parse in my head; I would expect something like `[[ "$(uname)" == "darwin" || "$(uname)" == "Darwin" ]]`.

Comment (majewsky) [reply golangvend:41]: Also, it can be made simpler like:
```
if hash sha256sum &>/dev/null; then
  # use sha256sum
else
  # use shasum
fi

**Comment by majewsky** [comment]
bump

**Comment by talal** [comment]
@majewsky Deriving from the original commit from Arno and your suggestion. The following works without any issues:

```
# macOS support
if hash sha256sum &>/dev/null; then
  sha256sum() {
    sha256sum "$@"
  }
else
  sha256sum() {
    shasum -a 256 "$@"
  }
fi
```
Also `find` is available on macOS by default (at least macOS 10.13) so there is no need for `find()` (line 40).

Kindly review and update.

**Comment by majewsky** [comment]
I added the fallback to `shasum -a 256` suggested by Talal, although in a slightly different way (29fb3fa).

Regarding `find` vs. `gfind`, I'm pretty sure I added this for a reason. Probably the stock `find` on macOS does not support everything I need. (Most of the coreutils are positively ancient on macOS. For example, run `man units` and look at the date in the bottom left corner. :smile:)

**Pull request closed without merging**